python/mlc_chat/rest.py

```diff
@@ -180,6 +184,69 @@ async def __anext__(self):
         raise StopAsyncIteration
 
 
+def add_function_call(prompt: List[ChatMessage], function_string: str):
+    # update content of the last input message to include function string
+    user_query = prompt[-1].content
+    prompt[-1].content = f"<<question>> {user_query} <<function>> {function_string}\n"
```

Comment:
Read a bit, looks like the `<<question>>` `<<function>>` here are specific to the gorilla model. This indicates that we may need to extend the conversation protocol to incorporate how user queries interact with functions.

cc @tqchen

Reply:
let us survey a few other models that support function call, then update template.